{"name": "test_button_check", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//button[@class=\"btn btn-danger btn-block\"]\"}\n  (Session info: chrome=125.0.6422.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n0   chromedriver                        0x0000000104fda4a0 chromedriver + 4301984\n1   chromedriver                        0x0000000104fd2de8 chromedriver + 4271592\n2   chromedriver                        0x0000000104c0419c chromedriver + 278940\n3   chromedriver                        0x0000000104c462c4 chromedriver + 549572\n4   chromedriver                        0x0000000104c7ec5c chromedriver + 781404\n5   chromedriver                        0x0000000104c3b004 chromedriver + 503812\n6   chromedriver                        0x0000000104c3b9ec chromedriver + 506348\n7   chromedriver                        0x0000000104fa24e8 chromedriver + 4072680\n8   chromedriver                        0x0000000104fa6f94 chromedriver + 4091796\n9   chromedriver                        0x0000000104f8972c chromedriver + 3970860\n10  chromedriver                        0x0000000104fa787c chromedriver + 4094076\n11  chromedriver                        0x0000000104f7c6ac chromedriver + 3917484\n12  chromedriver                        0x0000000104fc4ae0 chromedriver + 4213472\n13  chromedriver                        0x0000000104fc4c5c chromedriver + 4213852\n14  chromedriver                        0x0000000104fd29e0 chromedriver + 4270560\n15  libsystem_pthread.dylib             0x000000019e55af94 _pthread_start + 136\n16  libsystem_pthread.dylib             0x000000019e555d34 thread_start + 8", "trace": "browser = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"45b6727acff69b7f8a6178feae736c17\")>\n\n    @allure.feature('Button Check')\n    @allure.story('Check if the login button is displayed')\n    @allure.description('This test will check if the login button is displayed on the site')\n    def test_button_check(browser):\n        with allure.step('Open browser'):\n            authorization_page = AuthorizationPage(browser)\n    \n        with allure.step('Go to site'):\n            authorization_page.open()\n    \n        with allure.step('Handle notification if present'):\n            authorization_page.handle_notification()\n    \n        with allure.step('Login button is displayed'):\n            authorization_page.login_button()\n    \n        with allure.step('Enter button is displayed'):\n>           assert authorization_page.enter_button().is_displayed()\n\ntests/test_1_authorization.py:58: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/authorization_page.py:45: in enter_button\n    return self.find(enter_button)\npages/base_page.py:6: in find\n    return self.browser.find_element(*args)\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x105585bb0>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...00019e55af94 _pthread_start + 136\\\\n16  libsystem_pthread.dylib             0x000000019e555d34 thread_start + 8\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//button[@class=\"btn btn-danger btn-block\"]\"}\nE         (Session info: chrome=125.0.6422.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       0   chromedriver                        0x0000000104fda4a0 chromedriver + 4301984\nE       1   chromedriver                        0x0000000104fd2de8 chromedriver + 4271592\nE       2   chromedriver                        0x0000000104c0419c chromedriver + 278940\nE       3   chromedriver                        0x0000000104c462c4 chromedriver + 549572\nE       4   chromedriver                        0x0000000104c7ec5c chromedriver + 781404\nE       5   chromedriver                        0x0000000104c3b004 chromedriver + 503812\nE       6   chromedriver                        0x0000000104c3b9ec chromedriver + 506348\nE       7   chromedriver                        0x0000000104fa24e8 chromedriver + 4072680\nE       8   chromedriver                        0x0000000104fa6f94 chromedriver + 4091796\nE       9   chromedriver                        0x0000000104f8972c chromedriver + 3970860\nE       10  chromedriver                        0x0000000104fa787c chromedriver + 4094076\nE       11  chromedriver                        0x0000000104f7c6ac chromedriver + 3917484\nE       12  chromedriver                        0x0000000104fc4ae0 chromedriver + 4213472\nE       13  chromedriver                        0x0000000104fc4c5c chromedriver + 4213852\nE       14  chromedriver                        0x0000000104fd29e0 chromedriver + 4270560\nE       15  libsystem_pthread.dylib             0x000000019e55af94 _pthread_start + 136\nE       16  libsystem_pthread.dylib             0x000000019e555d34 thread_start + 8\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException"}, "description": "This test will check if the login button is displayed on the site", "steps": [{"name": "Open browser", "status": "passed", "start": 1716136936006, "stop": 1716136936006}, {"name": "Go to site", "status": "passed", "start": 1716136936006, "stop": 1716136937641}, {"name": "Handle notification if present", "status": "passed", "start": 1716136937642, "stop": 1716136938505}, {"name": "Login button is displayed", "status": "passed", "start": 1716136938505, "stop": 1716136938509}, {"name": "Enter button is displayed", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//button[@class=\"btn btn-danger btn-block\"]\"}\n  (Session info: chrome=125.0.6422.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n0   chromedriver                        0x0000000104fda4a0 chromedriver + 4301984\n1   chromedriver                        0x0000000104fd2de8 chromedriver + 4271592\n2   chromedriver                        0x0000000104c0419c chromedriver + 278940\n3   chromedriver                        0x0000000104c462c4 chromedriver + 549572\n4   chromedriver                        0x0000000104c7ec5c chromedriver + 781404\n5   chromedriver                        0x0000000104c3b004 chromedriver + 503812\n6   chromedriver                        0x0000000104c3b9ec chromedriver + 506348\n7   chromedriver                        0x0000000104fa24e8 chromedriver + 4072680\n8   chromedriver                        0x0000000104fa6f94 chromedriver + 4091796\n9   chromedriver                        0x0000000104f8972c chromedriver + 3970860\n10  chromedriver                        0x0000000104fa787c chromedriver + 4094076\n11  chromedriver                        0x0000000104f7c6ac chromedriver + 3917484\n12  chromedriver                        0x0000000104fc4ae0 chromedriver + 4213472\n13  chromedriver                        0x0000000104fc4c5c chromedriver + 4213852\n14  chromedriver                        0x0000000104fd29e0 chromedriver + 4270560\n15  libsystem_pthread.dylib             0x000000019e55af94 _pthread_start + 136\n16  libsystem_pthread.dylib             0x000000019e555d34 thread_start + 8\n\n", "trace": "  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/tests/test_1_authorization.py\", line 58, in test_button_check\n    assert authorization_page.enter_button().is_displayed()\n  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/pages/authorization_page.py\", line 45, in enter_button\n    return self.find(enter_button)\n           ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/pages/base_page.py\", line 6, in find\n    return self.browser.find_element(*args)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/.venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/gulim/PycharmProjects/PythonPOM/ProjectV1/.venv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1716136938509, "stop": 1716136948546}], "attachments": [{"name": "stdout", "source": "c4575fa1-0ac9-4482-a4b4-cd1df67c5196-attachment.txt", "type": "text/plain"}], "start": 1716136936006, "stop": 1716136948551, "uuid": "917d5b4a-c7db-450a-9dd9-cab7d1b2dc19", "historyId": "09c370d2bfc6d2ea701bb8e4bc4b0a04", "testCaseId": "09c370d2bfc6d2ea701bb8e4bc4b0a04", "fullName": "tests.test_1_authorization#test_button_check", "labels": [{"name": "story", "value": "Check if the login button is displayed"}, {"name": "feature", "value": "Button Check"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_1_authorization"}, {"name": "host", "value": "Gulims-MacBook-Air.local"}, {"name": "thread", "value": "8294-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_1_authorization"}]}